dev-lang/perl/perl-5.30.3-r2.ebuild

```diff
@@ -313,28 +263,131 @@ src_prepare_dynamic() {
 	ln -s ${LIBPERL} libperl$(get_libname ) || die
 }
 
+# Copy a patch into the patch series
+# add_patch SRC_PATH DEST_NAME ['description'] ['bug'] ['bug']
+# - description is optional, but recommended
+# - all arguments after descriptions are bug URLs
+add_patch() {
+	local patchdir="${WORKDIR}/patches"
+	local infodir="${WORKDIR}/patch-info"
+	local src_name dest_name desc
+	src_name="$1"
+	dest_name="$2"
+	desc="$3"
+	shift; shift; shift;
+	einfo "Adding ${dest_name} to patch bundle"
+	cp "${src_name}" "${patchdir}/${dest_name}" || die "Couldn't copy ${src_name} to ${dest_name}"
+	if [[ -n "${desc}" ]]; then
+		printf "%s" "${desc}" > "${infodir}/${dest_name}.desc" || die "Couldn't write ${dest_name}.desc"
+	fi
+	if [[ $# -gt 0 ]]; then
+		printf "%s\n" "$@" > "${infodir}/${dest_name}.bugs" || die "Couldn't write ${dest_name}.bugs"
+	fi
+}
+# Remove a patch using a glob expr
+# eg:
+#	 rm_patch *-darin-Use-CC*
+#
+rm_patch() {
+	local patchdir="${WORKDIR}/patches"
+	local expr="$1"
+	local patch="$( cd "${patchdir}"; echo $expr )"
+	einfo "Removing $patch ($expr) from patch bundle"
+	if [[ -e "${patchdir}/${patch}" ]]; then
+		rm -f "${patchdir}/${patch}" || die "Can't remove ${patch} ( $expr )"
+	else
+		ewarn "No ${expr} found in ${patchdir} to remove"
+	fi
+}
+c_escape_string() {
+	local slash dquote
+	slash='\'
+	dquote='"'
+	re_slash="${slash}${slash}"
+	re_dquote="${slash}${dquote}"
+
+	# Convert \ to \\,
+	#         " to \"
+	echo "$1" |\
+		sed "s|${re_slash}|${re_slash}${re_slash}|g" |\
+		sed "s|${re_dquote}|${re_slash}${re_dquote}|g"
+}
```

Comment:
this is probably simpler if you use single quotes for the sed regexps (and you won't need the variables then)
also, sed supports more than one action
sed -e 's|\|\\|g' -e 's|"|\"|g'
(please test)

Reply:
The problem here is single quotes aren't enough. The single quotes only allow you to subvert bash escaping, not the fact that "\" has regex-specific behaviour. Took me a while to get this function to work >_>.

Expanded, it turns out that you're doing: 

    sed 's|\\|\\\\|g' 

And now you have to make sure you have exactly the right number of \'s manually, visually.

The second one is doing:

    sed 's|\"|\\\"|' 

I think. And again, it makes my eyes dizzy.  Probably could be better documented.

Reply:
Also, seems "\\" has github behaviour too, and it gets munched in comments. lol.

    "\"  => ""
    \\    => "\"